jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/DefaultToolkitConnectionManager.kt

```diff
@@ -108,7 +115,7 @@ class DefaultToolkitConnectionManager : ToolkitConnectionManager, PersistentStat
             if (oldConnection != null && newConnection != null && pinningManager != null) {
                 val featuresToPin = mutableListOf<FeatureWithPinnedConnection>()
                 FeatureWithPinnedConnection.EP_NAME.forEachExtensionSafe {
-                    if (!pinningManager.isFeaturePinned(it) && it.supportsConnectionType(oldConnection) && !it.supportsConnectionType(newConnection)) {
+                    if (!pinningManager.isFeaturePinned(it) && (oldConnection is ManagedBearerSsoConnection != newConnection is ManagedBearerSsoConnection)) {
```

Comment:
What does this change mean?

Reply:
I meant the oldConnection and newConnection is different type: IAM vs. bearerToken, pls let me know if there is something better I can use

Reply:
I guess I can also go with `it.supportsConnectionType(oldConnection) != it.supportsConnectionType(newConnection)`?